components/canvas/canvas_data.rs

```diff
@@ -568,8 +672,15 @@ impl<'a> CanvasData<'a> {
     pub fn set_transform(&mut self, transform: &Transform2D<f32>) {
         // If there is an in-progress path, store the existing transformation required
         // to move between device and user space.
-        if (self.path.is_some() || self.path_builder.is_some()) && self.path_to_device_space.is_none() {
-            self.path_to_device_space = Some(self.drawtarget.get_transform());
+        match self.path_state.as_mut() {
+            None |
+            Some(PathState::DeviceSpacePathBuilder(..)) => (),
+            Some(PathState::UserSpacePathBuilder(_, ref mut transform)) |
+            Some(PathState::UserSpacePath(_, ref mut transform)) => {
+                if transform.is_none() {
+                    *transform = Some(self.drawtarget.get_transform());
```

Comment:
So if you call `set_transform` twice in a row, only the first one will take effect on the `PathState` (because on the second one, the `transform` will not be `None` anymore). Is that correct behavior?

Reply:
This is correct. The transform that is stored is the transform that applies to all existing points on the path and needs to be applied to those points when creating the device space path builder.